llmfoundry/callbacks/async_eval_callback.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2023 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+import logging
+import os
+from typing import Any, Dict, Optional, Union
+
+from composer.core import Callback, Event, State, Time
+from composer.loggers import Logger
+from composer.loggers.mosaicml_logger import (MOSAICML_PLATFORM_ENV_VAR,
+                                              RUN_NAME_ENV_VAR)
+from composer.utils import create_interval_scheduler, dist
+from mcli.api.runs import ComputeConfig  # TODO: should be available in root
+
+from mcli import Run, RunConfig, create_run, get_run
+
+log = logging.getLogger(__name__)
+
+MAX_RUN_NAME_LENGTH = 40
+
+# Note: train parameter names. See comments if they are different from eval
+REQUIRED_PARAMS_FOR_EVAL = {
+    'device_eval_batch_size',
+    'icl_tasks',  # only required for eval
+    'max_seq_len',
+    'model',  # models
+    'save_folder',  # required, but used as load_path
+}
```

Comment:
Is it possible to set which version of foundry you want to run with which may affect this?

Can we add a test to ensure this stays in-sync?

Reply:
For now, could we require the eval and training run require the same foundry version? Otherwise this gets a little hairy, this callback would need to be backwards compatible for all eval and train param changes

Reply:
yeah i think its fair to require same version